dlt/transformations/decorators.py

```diff
@@ -0,0 +1,139 @@
+from typing import Callable, Any, overload, Optional, Type
+
+from dlt.common.utils import get_callable_name
+from dlt.common.destination.dataset import SupportsReadableDataset
+from dlt.common.typing import AnyFun, Concatenate, TColumnNames
+from dlt.extract.incremental import TIncrementalConfig
+
+from dlt.common.schema.typing import (
+    TWriteDisposition,
+    TTableSchemaColumns,
+    TSchemaContract,
+    TTableFormat,
+    TTableReferenceParam,
+)
+
+from dlt.transformations.typing import (
+    TTransformationType,
+    TTransformationFunParams,
+    TLineageMode,
+)
+from dlt.transformations.transform import make_transform_resource, DltTransformResource
+from dlt.transformations.configuration import TransformConfiguration
+
+
+# NOTE: can we just return a resource directly with some additional hints here?
+@overload
+def transformation(
+    func: None = ...,
+    /,
+    name: str = None,
+    table_name: str = None,
+    write_disposition: TWriteDisposition = None,
+    columns: TTableSchemaColumns = None,
+    primary_key: TColumnNames = None,
+    merge_key: TColumnNames = None,
+    schema_contract: TSchemaContract = None,
+    table_format: TTableFormat = None,
+    references: TTableReferenceParam = None,
+    selected: bool = True,
+    spec: Type[TransformConfiguration] = None,
+    parallelized: bool = False,
+    incremental: Optional[TIncrementalConfig] = None,
+    standalone: bool = False,
+    # transformation specific
+    transformation_type: TTransformationType = None,
+    chunk_size: int = None,
+    lineage_mode: TLineageMode = None,
+) -> Callable[
+    [Callable[Concatenate[SupportsReadableDataset[Any], TTransformationFunParams], Any]],
```

Comment:
You do not need Concatenate here, you are not hiding first argument in the signature

Reply:
yeah true, this is a leftover. removed it!